web/src/lib/components/shared-components/search-bar/search-people-section.svelte

```diff
@@ -38,15 +40,23 @@
     }
     selectedPeople = selectedPeople;
   }
+
+  const filterPeople = (list: PersonResponseDto[], name: string) => {
+    return name == '' ? list : list.filter((p) => p.name.toLowerCase().startsWith(name.toLowerCase()));
+  };
 </script>
 
 {#await peoplePromise then people}
   {#if people && people.length > 0}
-    {@const peopleList = showAllPeople ? people : people.slice(0, numberOfPeople)}
+    {@const peopleList = showAllPeople ? filterPeople(people, name) : people.slice(0, numberOfPeople)}
 
     <div id="people-selection" class="-mb-4">
-      <div class="flex items-center gap-6">
-        <p class="immich-form-label">PEOPLE</p>
+      <div class="flex items-center w-full justify-between gap-6">
+        <p class="immich-form-label py-3">PEOPLE</p>
+
+        {#if showAllPeople}
+          <SearchBar bind:name placeholder="Filter people" isSearching={false} />
```

Comment:
The searchbar contains a `<button>` for the search icon and clicking it will submit the form. Maybe add `type="button"` to prevent that

Reply:
I see that behavior now. Adding `type="button"` didn't prevent it. Also tried adding `on:search={(event) => event.stopPropagation()})`. Will keep looking

Reply:
Strange, for me it does work when I add type="button" here. Maybe you added it to `<SearchBar>` directly?

https://github.com/immich-app/immich/blob/321525ead593e8b270741378cde912a5bb55bb27/web/src/lib/components/elements/search-bar.svelte#L26

Reply:
I did indeed try adding it to `<SearchBar>` :)

Reply:
The dispatch isn't relevant actually, because we don't listen for it and custom events don't automatically bubble up. The behavior you're seeing is caused by the default type=submit that buttons have and by putting it inside `<form>` it will submit the form when you click it. Adding type=button disables the default behavior.

Reply:
Ah, makes sense, thanks for the explanation.